client/my-sites/exporter/exporter.jsx

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import CompactCard from 'components/card/compact';
+import Gridicon from 'components/gridicon';
+import Button from 'components/forms/form-button';
+import AdvancedOptions from 'my-sites/exporter/advanced-options';
+
+export default React.createClass( {
+	displayName: 'Exporter',
+
+	render: function() {
+		return (
+			<div className="section-export">
+				<CompactCard>
+					<header>
+						<Button
+							className="exporter__export-button"
+							disabled={ false }
+							isPrimary={ true }
+						>
+							{ this.translate( 'Export' ) }
+						</Button>
+						<h1 className="exporter__title">
+							{ this.translate( 'Download an Export File' ) }
+						</h1>
+					</header>
+					<a href="#" onClick={ this.props.toggleAdvancedSettings }>
+						<Gridicon
+							icon={ this.props.advancedSettings.isVisible ? 'chevron-up' : 'chevron-down' }
+							size={ 16 } />
+						{ this.translate( 'Advanced Export Settings' ) }
+					</a>
+				</CompactCard>
+
+				{
+					this.props.advancedSettings.isVisible &&
+					<AdvancedOptions
```

Comment:
I might dissent a bit here. While I'd agree that it doesn't seem important that the data components necessarily live at the top of the component hierarchy, I do like the idea of a hard rule to keep data and presentational components separate.

You may already be familiar with these articles (and they're linked in the docs mentioned above); they do well to highlight the need for and purpose of splitting container components from their visual counterpart:

- https://medium.com/@learnreact/container-components-c0e67432e005
- https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0

My main takeaways here are that the following gains are to be had by always keeping "smart" and "dumb" components separate:

- __Reusability:__ Separating data from presentation enables each to be portable. The visual component can be rendered in any context so long as its data needs are satisfied, regardless of those source of that data.
- __Clarity:__ If merged, I'd be wasting effort deciphering where the data management ends and the visual handling begins. Single-purpose components tend to be smaller and easier to digest. We shouldn't be afraid to create and nest components.
- __Development ease:__ Simple-to-follow rules can eliminate the time a developer would otherwise spend evaluating when and how to separate the two. Predictable patterns enable outside developers to quickly grok the purpose of the components.

(The grain-of-salt to take here is that I have a tendency to be quick to abstract, though I've tried my best to account for [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) in the reasoning above.)

The issue I have with the implementation here is that we should do our best to be as granular as possible with what we're sending down to child components. In hind-sight, this is something that I had wished we'd done a better job of managing props in the post editor (ref: p4TIVU-11B-p2). Being more granular and making a best effort to send non-complex props makes it much easier to achieve pure components, improves reusability, and clearly identifies the needs of the component. Rather than passing a large `advancedSettings` object through the hierarchy here, I think it'd be better to pick off and pass only the required data in rendering each component.

Related docs:

- https://facebook.github.io/react/docs/thinking-in-react.html#step-3-identify-the-minimal-but-complete-representation-of-ui-state
- https://facebook.github.io/react/docs/pure-render-mixin.html